packages/back-end/src/events/handlers/slack/handleFeatureEventForSlack.ts

```diff
@@ -0,0 +1,312 @@
+): string[] => {
+  return getEnabledEnvironmentsForEvent(featureEvent);
+};
+
+/**
+ * Returns a list of the environments that are enabled for the event.
+ * For events with multiple states (e.g. "feature.updated"), it will include environments
+ * that are enabled in any of the available states (e.g. both `previous` and `current`)
+ * @param featureEvent
+ */
+const getEnabledEnvironmentsForEvent = (
+  featureEvent:
+    | FeatureCreatedNotificationEvent
+    | FeatureUpdatedNotificationEvent
+    | FeatureDeletedNotificationEvent
+): string[] => {
+  switch (featureEvent.event) {
+    case "feature.created":
+      return getEnabledEnvironmentsForEnvironmentSettings(
+        featureEvent.data.current.environmentSettings
+      );
+
+    case "feature.updated":
+      return uniq(
+        getEnabledEnvironmentsForEnvironmentSettings(
+          featureEvent.data.previous.environmentSettings
+        ).concat(
+          getEnabledEnvironmentsForEnvironmentSettings(
+            featureEvent.data.current.environmentSettings
+          )
+        )
+      );
+
+    case "feature.deleted":
+      return getEnabledEnvironmentsForEnvironmentSettings(
+        featureEvent.data.previous.environmentSettings
+      );
+  }
+};
+
+const getEnabledEnvironmentsForEnvironmentSettings = (
+  environmentSettings: Record<string, FeatureEnvironment>
+): string[] => {
+  if (!environmentSettings) {
+    return [];
+  }
+
+  return Object.keys(environmentSettings).filter(
+    (env) => environmentSettings[env]?.enabled
+  );
+};
+
+type BuildSlackMessageOptions = {
+  featureEvent:
+    | FeatureCreatedNotificationEvent
+    | FeatureUpdatedNotificationEvent
+    | FeatureDeletedNotificationEvent;
+  slackIntegration: SlackIntegrationInterface;
+  eventId: string;
+};
+/**
+ * Given an event, will build the desired Slack message
+ * @param featureEvent
+ * @param slackIntegration
+ * @param eventId
+ */
+const buildSlackMessageForEvent = ({
+  featureEvent,
+  slackIntegration,
+  eventId,
+}: BuildSlackMessageOptions): SlackMessage => {
+  switch (featureEvent.event) {
+    case "feature.created":
+      return buildCreatedEvent(featureEvent, slackIntegration, eventId);
+
+    case "feature.updated":
+      return buildUpdatedEvent(featureEvent, slackIntegration, eventId);
+
+    case "feature.deleted":
+      return buildDeletedEvent(featureEvent, slackIntegration, eventId);
+  }
+};
+
+const getIntegrationContextBlock = (
+  slackIntegration: SlackIntegrationInterface
+): KnownBlock => {
+  return {
+    type: "context",
+    elements: [
+      {
+        type: "plain_text",
+        text: `This was sent from your Slack integration: ${slackIntegration.name}`,
+      },
+    ],
+  };
+};
+
+const getFeatureUrlFormatted = (featureId: string): string =>
+  `\n• <${APP_ORIGIN}/features/${featureId}|View Feature>`;
+
+const getEventUrlFormatted = (eventId: string): string =>
+  `\n• <${APP_ORIGIN}/events/${eventId}|View Event>`;
```

Comment:
It doesn't seem like we can format bullets with markdown from the API so i'm making a bulleted list here with the • character.